BPA/Ch.10/js_rest_occ.stan

```diff
@@ -186,45 +204,43 @@ generated quantities {
     int q[n_occasions - 1];
     real mu2;
 
-    z[i, 1] <- bernoulli_rng(gamma[1]);
+    z[i, 1] = bernoulli_rng(gamma[1]);
     for (t in 2:n_occasions) {
-      q[t - 1] <- 1 - z[i, t - 1];
-      mu2 <- phi[i, t - 1] * z[i, t - 1] +
-             gamma[t] * prod(q[1:(t - 1)]);
-      z[i, t] <- bernoulli_rng(mu2);
+      q[t - 1] = 1 - z[i, t - 1];
```

Comment:
This looks like that same kind of loop again, so maybe the function could be reused?  Or something like it to avoid the `N^2` operations?